hoomd/hpmc/update.py

```diff
@@ -512,6 +513,207 @@ def N(self):  # noqa: N802 - allow N as a function name
         return N_dict
 
 
+class Shape(Updater):
+    """Apply shape updates to the shape definitions defined in the integrator.
+
+    Args:
+        trigger (Trigger): Call the updater on triggered time steps.
+
+        step_size (float): Default maximum size of shape trial moves.
+
+        pretend (bool, optional): When True the updater will not actually update
+            the shape definitions. Instead, moves will be proposed and the
+            acceptance statistics will be updated correctly.
+
+        nselect (int, optional): Number of types to change every time the
+            updater is called.
+
+        nsweeps (int, optional): Number of times to update shape definitions
+            during each triggered timesteps.
+
+        multi_phase (bool, optional): When True MPI is enforced and shapes are
+            updated together for two boxes.
+
+        num_phase (int, optional): How many boxes are simulated at the same
+            time, supported values are 1, 2 and 3.
+
+    Examples::
+
+        mc = hoomd.hpmc.integrate.ConvexPolyhedron(23456)
+        mc.shape["A"] = dict(vertices=numpy.asarray([(1, 1, 1), (-1, -1, 1),
+                                                    (1, -1, -1), (-1, 1, -1)]) / 2)
+        vertex_move = hoomd.hpmc.shape_move.Vertex(stepsize={'A': 0.01},
+                                                   param_ratio=0.2,
+                                                   volume=1.0)
+        updater = hoomd.hpmc.update.Shape(shape_move=vertex_move,
+                                          trigger=hoomd.trigger.Periodic(1),
+                                          nselect=1,
+                                          nsweeps=1,
+                                          multi_phase=False,
+                                          num_phase=1)
+
+    Attributes:
+        trigger (Trigger): Call the updater on triggered time steps.
+
+        step_size (`TypeParameter` [``particle type``, `float`]): Maximum size
+            of shape trial moves.
+
+        shape_move (ShapeMove): Type of shape move to apply when updating shape
+            definitions
+
+        pretend (bool): When True the updater will not actually update the shape
+            definitions, instead moves will be proposed and the acceptance
+            statistics will be updated correctly.
+
+        nselect (int): Number of types to change every time the updater is
```

Comment:
`nselect` here is quite different that it is for the `HPMCIntegrator` class. I don't have a better name in my head, but I wanted to mention it in case someone else did.

Reply:
This variable is named `m_type_select` in C++, perhaps something like that?

Reply:
I like that!